rio/subsystems/drivetrain.py

```diff
@@ -152,21 +201,51 @@ def setModuleState(self, newState):
         the "state" (steering and speed)
         of a module.
         """
-
         # TODO: Use optimization at some point
-        self.state = newState
+
+        optimizedState = kinematics.SwerveModuleState.optimize(
+            newState, self.getModuleState().angle
+        )
+
+        if optimizedState.angle.radians() != newState.angle.radians():
+            print("Optimized!")
+
+        currentRef = (optimizedState.angle.radians() / (2 * math.pi)) * self.pid[
+            "steer"
+        ]["ratio"]
```

Comment:
The issue may be here, we never update the reference beyond 1 rotation so we can never continuously rotate the wheel around and around!